lib/action_view/component/preview.rb

```diff
@@ -19,8 +27,8 @@ def all
         end
 
         # Returns the arguments for rendering of the component in its layout
-        def render_args(example)
-          new.public_send(example).merge(layout: @layout)
+        def render_args(example, params: {})
+          new(params: params).public_send(example).merge(layout: @layout)
```

Comment:
Instead of making params a property of the Preview my not make them arguments of the example?

```ruby
new.public_send(example, **params).merge(layout: @laout)
```

Reply:
Hey Jon, thank you for your feedback. I really look forward to see your work related to Storybook, that seems great! :grinning: 

I've been thinking a bit about your proposed change.

On one hand I like it because it simplifies the access of params from the example: you just define them as method arguments and it'd set them from the params if they're given.

The implementation would be a little bit different though, as we receive `ActionController::Parameters` from the controller, which would include also other data (`controller`, `action`, etc), so we need to only take the defined keywords in order to avoid passing unsupported ones:

```ruby
# components_controller.rb
@render_args = @preview.render_args(@example_name, params: params.permit!)

# preview.rb
def render_args(example, params: {})
  method_params = instance_method(example).parameters.map(&:last)
  provided_params = params.slice(*method_params).to_h.symbolize_keys
  new.public_send(example, **provided_params).merge(layout: @layout)
end

# component_preview.rb
def default(cta: "Click me!", title: "Lorem Ipsum")
```

On the other hand there are a couple of things that I'm not 100% sure about this approach:

1. We have some extra logic to make it working, which could break. What happens if we define on the method something else than keywords? Eg. `def default(some_config = nil, title: "Lorem Ipsum")`, we'd then may have to take care of that case.

2. I'm trying to think about someone who is adding the example and what they may think if they want to add a value from url parameters. Would they expect to have them passed when they set a keyword argument, or would they try to access the params directly? I mean, if you're on Rails it's very likely that you know that in a controller you can access them by calling `params`, so I'm wondering if there would be a similar expectation or not :thinking: 

What do you think?

Reply:
@cover I certainly share you concerns. We likely will also find problems calling the example with zero arity vs positive arity (we already trie to solve this in constructors for Ruby 2.7 that I think we rolled back)

I could go either way, knowing to check params is probably fine as as you point out similar to Rails controller. In all honesty its challenges like this that in part inspired me to go with the DSL I previewed for declaring Stories in Ruby. IMO the explicitness of what the "params" are feels more natural to me.

Reply:
Cool, I'll update this to pass down the params through the method/example arguments (pretty much the code above) once the current library renaming is merged :+1: